packages/code/src/CopyButton/CopyButton.tsx

```diff
@@ -39,31 +51,70 @@ function CopyButton({ onCopy, contents }: CopyProps) {
     }
 
     return () => clipboard.destroy();
-  }, [buttonNode, contents, copied, portalContainer]);
+  }, [buttonRef, contents, copied, portalContainer]);
 
   const handleClick = (e: React.MouseEvent) => {
     e.preventDefault();
+    onCopy?.();
+    setCopied(true);
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    switch (e.key) {
+      case keyMap.Escape:
+      case keyMap.Tab: {
+        closeTooltip();
+        break;
+      }
 
-    if (onCopy) {
-      onCopy();
+      case keyMap.Enter:
+      case keyMap.Space: {
+        e.preventDefault();
+        buttonRef.current?.click();
```

Comment:
Does clicking enter not automatically trigger a click?

Reply:
Yeah for some strange reason, it doesn't. I originally tried invoking `handleClick`. It would trigger the state change but wouldn't create a clipboard of the contents. I ended up opting for this approach

Reply:
Is it possible to test that the correct content is copied?

Reply:
ahh I thought this was already covered in the original code spec file but I guess it's not. I will add to this spec file

Reply:
no worries, I also assumed the same thing. I'm not sure why its missing.